src/ai/backend/manager/models/vfolder.py

```diff
@@ -814,24 +833,46 @@ async def load_slice(
         filter: str = None,
         order: str = None,
     ) -> Sequence[VirtualFolder]:
-        from .group import groups
+        from .group import groups, query_user_accessible_groups
         from .user import users
 
-        j = vfolders.join(users, vfolders.c.user == users.c.uuid, isouter=True).join(
-            groups, vfolders.c.group == groups.c.id, isouter=True
+        j = (
+            sa.join(
+                vfolders,
+                vfolder_permissions,
+                vfolders.c.id == vfolder_permissions.c.vfolder,
+                isouter=True,
+            )
+            .join(users, vfolders.c.user == users.c.uuid, isouter=True)
+            .join(groups, vfolders.c.group == groups.c.id, isouter=True)
```

Comment:
Same here.

Reply:
The original code make 3 joins and I did not add or remove any join here because 3 joins does not seem to have bad performance.
How about adding index on users email column and groups name column for performance on this query?